src/main/java/org/spongepowered/api/event/advancement/CriterionRevokeEvent.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.advancement;
+
+import org.spongepowered.api.event.Cancellable;
+
+public interface CriterionRevokeEvent extends CriterionEvent, Cancellable {
```

Comment:
I feel like elsewhere in the API we keep these as sub-interfaces to the original since they're so similar and it seems generally the `CriterionEvent` wouldn't be used on its own. Is there a particular reason we shouldn't here? 

For example look at `TabCompleteEvent` or `RconConnectionEvent` and its sub-interfaces. 

So if we make this change you'd listen to `CriterionEvent.Revoke` for this one. 

This would also apply to the advancement events.